vowpalwabbit/plt.cc

```diff
@@ -0,0 +1,382 @@
+// Copyright (c) by respective owners including Yahoo!, Microsoft, and
+// individual contributors. All rights reserved. Released under a BSD (revised)
+// license as described in the file LICENSE.
+#include <cfloat>
+#include <cmath>
+#include <cstdio>
+#include <sstream>
+#include <vector>
+#include <unordered_map>
+#include <unordered_set>
+#include <algorithm>
+#include <queue>
+#include "reductions.h"
+#include "vw.h"
+
+using namespace VW::LEARNER;
+using namespace VW::config;
+
+namespace plt_ns
+{
+struct node
+{
+  uint32_t n;  // node number
+  float p;     // node probability
+
+  bool operator<(const node& r) const { return p < r.p; }
+};
+
+struct plt
```

Comment:
Just an FYI for the future.  We started adding learn, predict methods to these structs and creating unit tests to test the basics of learn/predict.  Perhaps in a future PR?

Reply:
I can change it in feature PR, can you provide an example of already modified learner?

Reply:
not sure if there's a better example regarding the learn/predict as part of struct https://github.com/VowpalWabbit/vowpal_wabbit/blob/master/vowpalwabbit/cb_adf.cc#L473

Regarding the tests, not sure which unit tests Rajan is referencing.